### ISSUE TITLE
Polish VS Code messages in the terminal and make consistent

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#150858]
Looking at https://tabby.sh/about/features I really like how consistent all the terminal-specific messages are and how great they look compared to our (@eugeny 🥇). I think we should do a pass over all the messages we write to the terminal and make them consistent. I think this is everything currently:

The inverting the whole row thing didn't end up working well, and does anyone care about the restore time?

![image](https://user-images.githubusercontent.com/2193314/171194457-621b37f5-ed5b-44b3-836f-8b4b97ea5669.png)

Tasks:

![image](https://user-images.githubusercontent.com/2193314/171194602-622d313f-eac0-439b-8ead-2762fbfbe2c4.png)

Local shell in remote workspace:

![image](https://user-images.githubusercontent.com/2193314/171194910-e3163efc-7273-4dd0-a235-68f588fba218.png)

We removed the recent shell integration message.

Maybe inverting and using ℹ️ ⚠️? We could also write "VSC"/"VS Code" and/or have a hover with more info when hovering the icon?

Thoughts @misolori @daviddossett

**Comment by daviddossett** [comment]
Got a screenshot of the terminal-specific messages in Tabby? 

The inverted text always feels quite heavy to me. Using icons could look something like this:

<img width="1194" alt="CleanShot 2022-05-31 at 10 46 08@2x" src="https://user-images.githubusercontent.com/25163139/171251670-ce53d444-49c8-49bd-a988-0d3ca85a9c19.png">

or with emojis like..

<img width="1221" alt="CleanShot 2022-05-31 at 10 46 14@2x" src="https://user-images.githubusercontent.com/25163139/171251628-1c8ef04c-2324-41cf-a553-b86a670fe957.png">

**Comment by daviddossett** [comment]
Oh ok, that's inversion actually looks quite nice. I especially like the different colors involved. I'm sure the colors/contrast needs to be tweaked here but we could do something along those lines:

<img width="1230" alt="CleanShot 2022-05-31 at 11 22 49@2x" src="https://user-images.githubusercontent.com/25163139/171257963-59fd36a9-9faa-4d3c-ac09-7b4314a1fd1e.png">

**Comment by Tyriar** [comment]
Yeah I like the invert on the left and a more toned down color after. Not sure if we can guarantee the warning/info icons being outlines or emoji atm, and we probably couldn't use the codicon symbols without some hacks.

**Comment by misolori** [comment]
Another idea is to have some type of "badge" next to the line that can give more info if we have (ssh/task/etc). Especially when it's terminal commands that are chained up, we want to make sure it's too noisy:

<img width="785" alt="CleanShot 2022-05-31 at 11 35 13@2x" src="https://user-images.githubusercontent.com/35271042/171260032-9f51e0ea-50e3-43c6-8c1a-5fff599ed513.png">

**Comment by Eugeny** [comment]
If the badge could be an xterm decoration that's actually on the left margin, it would make it unambiguous that it's actually coming from VSCode itself and is not just some terminal output.

**Comment by Tyriar** [comment]
@Eugeny good idea, we could use codicons then too 🤔

**Comment by Tyriar** [comment]
Similar styling to tabby in context using a simple invert followed by ansi white on ansi blue:

<img width="511" alt="Screen Shot 2022-06-10 at 10 14 52 am" src="https://user-images.githubusercontent.com/2193314/173117897-d0d4064a-b70f-443b-904a-0b9dea1a7203.png">
<img width="544" alt="Screen Shot 2022-06-10 at 10 15 20 am" src="https://user-images.githubusercontent.com/2193314/173117940-6fc83b83-3395-414f-bb2b-8a9ec354a615.png">

---

<img width="638" alt="Screen Shot 2022-06-10 at 10 23 43 am" src="https://user-images.githubusercontent.com/2193314/173119060-8295e9ca-199d-4b29-9b67-e0e24547a82d.png">
<img width="644" alt="Screen Shot 2022-06-10 at 10 23 29 am" src="https://user-images.githubusercontent.com/2193314/173119015-0235f91c-6bcb-4079-8ec3-867c6d744996.png">

---
<img width="447" alt="Screen Shot 2022-06-10 at 10 24 36 am" src="https://user-images.githubusercontent.com/2193314/173119198-3fdfe4ed-e67d-4077-a5f0-93e9fd963749.png">
<img width="424" alt="Screen Shot 2022-06-10 at 10 24 46 am" src="https://user-images.githubusercontent.com/2193314/173119225-c8ecd0e4-0e60-4b4d-ad8a-18c0b3e88e7c.png">

Here's just the inverted part and default background for the rest:

<img width="401" alt="Screen Shot 2022-06-10 at 10 27 36 am" src="https://user-images.githubusercontent.com/2193314/173119690-8630e4e5-eeb2-40b7-885b-b8e365d6aa51.png">

I think the more prominent background color is preferable here since these messages are used relatively infrequently, important and warrant differentiating like that (currently warnings or splitting the buffer up into chunks), compared to @misolori's more subtle approach. If opinions are split on this we could also have a setting with either a set of styles (eg. `"loud" | "subtle"`or allowing specifying the actual SGR style sequence (eg. `"1;34"` = bold blue foreground).

As for making those lines more special we can do a few things with the help of decorations:

- Overlay the `*` with a warning/info codicon or a dot. It's probably best to keep the * as a fallback underneath in case it somehow makes it to another terminal
- Give it hover feedback
- Give it a hover. For example, when the session was saved, link to docs, etc. This could apply to just the * or the whole line
- Have it outside the character grid, this would certainly help differentiate but I'm not sure it's worth it in this case?
- Give it rounded corners, I'm a little concerned about it not fitting in with the rest of the terminal

**Comment by misolori** [comment]
I like the simplicity of using `*`. Personally, I would love a setting to control this and wouldn't want something so strong.

**Comment by Tyriar** [comment]
There are also these messages:

![image](https://user-images.githubusercontent.com/2193314/173341524-6e02d2c9-8b81-4ad6-a912-7c3ab933a057.png)

**Comment by Tyriar** [comment]
Positive feedback in UX sync, will keep my eyes open for feedback.

**Comment by Tyriar** [comment]
@kamenminkov there's no option to customize currently, I was trying to avoid adding _yet another_ setting for something so small. What sort of formatting options were you after?

**Comment by kamenminkov** [comment]
@Tyriar I was kind of used to the bold font of that message. If I could customise the messages, I'd probably make them bold again. What's shown by @misolori above also looks pretty good to me. I guess I could get used to the new look.

**Comment by Tyriar** [comment]
Hmm ok. I'll keep an eye out for more feedback, sounds like you're just used to something else. I wanted to avoid using the loud formatting as that would probably get distracting:

![image](https://user-images.githubusercontent.com/2193314/174113036-cf4db183-b0c6-404a-aec9-1b6d2293acfc.png)

We'll also be having "command decorations" coming to tasks soon which will further differentiate when tasks _start_:

![image](https://user-images.githubusercontent.com/2193314/174113240-d6184ccb-e882-469a-a5a5-8df60443bdfa.png)

**Comment by gdh1995** [comment]
Hello today I noticed VS Code v1.69.0 shows such a blue background color for terminal messages like `History restored`. But I can not find where I can modiify its color value.

On my VS Code, it looks like hard to read (My theme is `Light+ (default light)`):
![image](https://user-images.githubusercontent.com/5547703/177942519-881d18c8-1992-40ac-b775-abde79f045de.png)

I wonder how to modify the background / foreground color of such messages.

BTW, I'm using Windows 10+11, but since a recent version, VS Code terminal doesn't obey the colors defined in `HKEY_CURRENT_USER\Console`'s `ColorTableXX` items. How can I change it back?

**Comment by Tyriar** [comment]
@gdh1995 looks like you disabled the minimum contrast ratio feature which ensures that is readable. Changing this to something else will make it not readable in other themes which is why we enable the feature by default now; it's not possible to ensure a good contrast ratio without it. So I'd recommend tweaking `terminal.integrated.minimumContrastRatio` to be higher than 1.

Here's 2.5:

![image](https://user-images.githubusercontent.com/2193314/177988571-31fda8fe-4bff-4a1a-80c4-588bee2be11b.png)

And 3:

![image](https://user-images.githubusercontent.com/2193314/177988606-b2628d37-1df1-4c7b-81f1-b421c18f0c55.png)

I also created https://github.com/microsoft/vscode/issues/154500 to make blue and bright blue different as they're currently the same in Light+

**Comment by gdh1995** [comment]
@Tyriar Thank you. It's indeed about `terminal.integrated.minimumContrastRatio`. I once turned off it because it makes `blue` and `light blue` in my customized color table not easy to distinguish, so today it's time to find some other colors.

**Comment by Tyriar** [comment]
@gdh1995 if you prefer to control everything tightly you can also keep min contrast ratio off and customize the individual colors in your settings.json:

```json
  "workbench.colorCustomizations": {
    "[Default Light+]": {
      "terminal.ansiBlue": "#0000ff",
    }
  }
```

The only problem with that is you will inevitably hit programs that cause contrast problems.
